### PR TITLE
Fixes filesystem provider details

### PR DESCRIPTION
**Pull request by vbnrh**
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2228816

This commit fixes the provider details in namespacestore details page.

Screenshot 2023-08-08 at 1 31 22 PM

### CODE REVIEW

File: packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ const SystemCapacityCard: React.FC = () => {
   return (
     <Card className="odf-capacityCard--height">
       <CardHeader>
-        <CardTitle>{t('System capacity')}</CardTitle>
```
Comment (SanjalKatiyar) [inline packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx:90]: let's keep this as part of its corresponding BZ... this change has nothing to do with this PR...

File: locales/en/plugin__odf-console.json [hunk 2]
```diff
@@ -893,6 +893,7 @@
   "NamespaceStore details": "NamespaceStore details",
   "Target Blob Container": "Target Blob Container",
   "Num Volumes": "Num Volumes",
+  "PVC Name": "PVC Name",
```
Comment (SanjalKatiyar) [inline locales/en/plugin__odf-console.json:896]:
```suggestion
  "PVC name": "PVC name",
```

File: packages/odf/constants/mcg.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export enum BC_PROVIDERS {
   GCP = 'Google Cloud Storage',
   AZURE = 'Azure Blob',
   IBM = 'IBM COS',
-  FILESYSTEM = 'Filesystem',
+  FILESYSTEM = 'nsfs',
```
Comment (SanjalKatiyar) [inline packages/odf/constants/mcg.ts:10]: won't this affect what we show on the dropdown as well ??
if we need to show `nsfs` as the dropdown item we need to confirm this with UX and also capitalise it...

File: packages/odf/components/resource-pages/Providers.tsx [hunk 4]
```diff
@@ -181,6 +181,28 @@ const PVCDetails: React.FC<ProviderDetailsProps> = ({ resource }) => {
...
+
+  return (
+    <Flex direction={{ default: 'column' }} className="details-item--border">
+      <FlexItem>
+        <DetailsItem field={t('Provider')}>
+          {BC_PROVIDERS.FILESYSTEM}
+        </DetailsItem>
+      </FlexItem>
+      <FlexItem>
+        <DetailsItem field={t('PVC Name')}>{pvcName}</DetailsItem>
```
Comment (SanjalKatiyar) [inline packages/odf/components/resource-pages/Providers.tsx:197]:
```suggestion
        <DetailsItem field={t('PersistentVolumeClaim')}>{pvcName}</DetailsItem>
```

Comment (SanjalKatiyar) [reply packages/odf/components/resource-pages/Providers.tsx:197]: basically making it same as what's displayed on the form...

File: packages/odf/components/resource-pages/Providers.tsx [hunk 5]
```diff
@@ -181,6 +181,28 @@ const PVCDetails: React.FC<ProviderDetailsProps> = ({ resource }) => {
   );
 };
 
+const FilesystemDetails: React.FC<ProviderDetailsProps> = ({ resource }) => {
+  const { t } = useCustomTranslation();
+  const pvcName = (resource as NamespaceStoreKind).spec.nsfs.pvcName;
```
Comment (bipuladh) [inline packages/odf/components/resource-pages/Providers.tsx:186]: Allow ProviderDetailsProps to take a type generic 
ProviderDetailsProps<NamespaceStoreKind>
This should stop us from using `as` here and multiple other places.

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: packages/odf/constants/mcg.ts (hunk 3 above)
Comment (vbnrh) [reply packages/odf/constants/mcg.ts]: i am changing this back to Filesystem. The map has the correct value as nsfs.

File: packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx (hunk 1 above)
Comment (vbnrh) [reply packages/odf/components/odf-dashboard/system-capacity-card/capacity-card.tsx]: Got accidentally added before i stashed the other changes.

File: packages/odf/components/resource-pages/Providers.tsx (hunk 5 above)
Comment (vbnrh) [reply packages/odf/components/resource-pages/Providers.tsx]:
```
type ProviderDetailsProps = {
  resource: BackingStoreKind | NamespaceStoreKind;
};
```
This was already there so i am removing the type casting.

**Pull request merged**